### PR TITLE
Add sentinel_2_nrt indexing package

### PR DESCRIPTION
**Pull request by pindge**
- read variable configuration from airflow variable setting
- utility tool for ad-hoc execution
   - ows-update-ranges
   - cubedash-gen
- archiving process 
- pending discussion then add indexing process
still need cleaning up, fix __doc__, fix typo etc

### CODE REVIEW

**Comment by alexgleith** [comment]
What's the justification for splitting this out into so many files?

**Comment by pindge** [comment]
> What's the justification for splitting this out into so many files?

I am splitting archiving process and indexing process. `ows-update --views` takes forever to run, indexing should run by itself.

`ows-update --views` and `cube-dash gen` via subdag operator have a ad-hoc dag by itself, where by passing `{{ dag_run.conf.products }}` it can update products as defined, so we don't have to `kubectl exec -it pod bash` to run updates as needed.

because it is split into multiple parts, it makes sense to share a common config file and common image constant file.

indexing-Page-2

File: dags/env_var/infra.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+INDEXING_ROLE = Variable.get("indexing_role", "dea-dev-eks-orchestration")
+
+# secrets name available in processing namespace
+SECRET_AWS_NAME = Variable.get("secret_aws_name", "processing-aws-creds-dev")
+SECRET_EXPLORER_NAME = Variable.get("secret_explorer_name", "explorer-db")
+SECRET_OWS_NAME = Variable.get("secret_ows_name", "ows-db")
+
+# DB config
+DB_DATABASE = Variable.get("db_database", "ows")
+DB_HOSTNAME = Variable.get("db_hostname", "db-writer")
```
Comment (NikitaGandhi) [inline dags/env_var/infra.py:18]: please revise above config as per this to be consistent: https://bitbucket.org/geoscienceaustralia/datakube/pull-requests/807/fix-indexing-endpoint-and-deploy-sandbox/diff

**Review by NikitaGandhi — CHANGES_REQUESTED** [review]

File: dags/sentinel_2_nrt/utility_explorer_summary.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        task_id="run-cubedash-gen-refresh-stat",
+        subdag=explorer_refresh_stats_subdag(
+            DAG_NAME,
+            "run-cubedash-gen-refresh-stat",
+            DEFAULT_ARGS,
+            SET_REFRESH_PRODUCT_TASK_NAME,
+        ),
+    )
+
+    START = DummyOperator(task_id="start_explorer_refresh_stats")
```
Comment (omad) [inline dags/sentinel_2_nrt/utility_explorer_summary.py:100]: We don't need these START/END markers, and they slow things down. I feel guilty for starting the practice. I think in some cases they make things clearer, but they're really not necessary.

**Review by omad — DISMISSED** [review]
Nothing urgent, looks okay so long as it works and I haven't tested that. :)

File: dags/sentinel_2_nrt/utility_explorer_summary.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+DAG_NAME = "utility_explorer-refresh-stats"
+
+# DAG CONFIGURATION
+DEFAULT_ARGS = {
+    "owner": "Pin Jin",
+    "depends_on_past": False,
+    "start_date": datetime(2020, 6, 14),
+    "email": ["pin.jin@ga.gov.au"],
+    "email_on_failure": False,
```
[the remark this answers is not in the document]
Comment (pindge) [reply dags/sentinel_2_nrt/utility_explorer_summary.py:45]: ok, so maybe not for utility ones,  those are only to be used when manual runs are required. I will set true for the processing ones

File: dags/sentinel_2_nrt/utility_explorer_summary.py (hunk 2 above)
Comment (pindge) [reply dags/sentinel_2_nrt/utility_explorer_summary.py:100]: will remove

File: dags/sentinel_2_nrt/sentinel_2_nrt_archive.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        image=INDEXER_IMAGE,
+        arguments=ARCHIVE_BASH_COMMAND,
+        labels={"step": "ds-arch"},
+        name="datacube-dataset-archive",
+        task_id="archive-nrt-datasets",
+        get_logs=True,
+        is_delete_operator_pod=True,
+    )
+
+    OWS_UPDATE_EXTENTS = SubDagOperator(
```
[the remark this answers is not in the document]
Comment (pindge) [reply dags/sentinel_2_nrt/sentinel_2_nrt_archive.py:97]: I need to find out how to make it a function call, as I kind grabbed this from airflow example from github

File: dags/env_var/infra.py (hunk 1 above)
Comment (pindge) [reply dags/env_var/infra.py:18]: Fixed

**Review by NikitaGandhi — APPROVED** [review]

**Review by NikitaGandhi — APPROVED** [review]
LGTM

**Pull request merged**